### PR TITLE
Renamed BrowsingContext to WindowProxy in script.

### PR DESCRIPTION
**Pull request by asajeffrey**
Renamed `script::dom::BrowsingContext` to `script::dom::WindowProxy`.

The browsing context is mostly maintained in the constellation, not in script. It would be nice to rename `constellation::Frame` to `constellation::BrowsingContext`, but that will be very confusing if there are two `BrowsingContext` types.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes do not require tests because renamings aren't externally visible

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16845)

### CODE REVIEW

**Comment by asajeffrey** [comment]
cc @jdm @cbrewster

File: components/script/dom/bindings/codegen/Bindings.conf [hunk 1]
```diff
@@ -34,8 +34,7 @@ DOMInterfaces = {
 },
 
 'WindowProxy' : {
-    'nativeType': 'BrowsingContext',
-    'path': 'dom::browsingcontext::BrowsingContext',
+    'path': 'dom::windowproxy::WindowProxy',
```
Comment (jdm) [inline components/script/dom/bindings/codegen/Bindings.conf]: This probably isn't needed any more.

**Review by jdm — CHANGES_REQUESTED** [review]

**Comment by jdm** [comment]
Otherwise this looks good to me.

File: components/script/dom/bindings/codegen/Bindings.conf (hunk 1 above)
Comment (asajeffrey) [inline components/script/dom/bindings/codegen/Bindings.conf]: I tried removing it and got:
```
error[E0432]: unresolved import `dom::types::WindowProxy`
   --> /home/ajeffrey/github/asajeffrey/servo/target/debug/build/script-dcc60cde0df0c3a0/out/Bindings/DissimilarOriginWindowBinding.rs:339:5
    |
339 | use dom::types::WindowProxy;
    |     ^^^^^^^^^^^^^^^^^^^^^^^ no `WindowProxy` in `dom::types`

error[E0432]: unresolved import `dom::types::WindowProxy`
   --> /home/ajeffrey/github/asajeffrey/servo/target/debug/build/script-dcc60cde0df0c3a0/out/Bindings/HTMLIFrameElementBinding.rs:342:5
    |
342 | use dom::types::WindowProxy;
    |     ^^^^^^^^^^^^^^^^^^^^^^^ no `WindowProxy` in `dom::types`

error[E0432]: unresolved import `dom::types::WindowProxy`
   --> /home/ajeffrey/github/asajeffrey/servo/target/debug/build/script-dcc60cde0df0c3a0/out/Bindings/WindowBinding.rs:647:5
    |
647 | use dom::types::WindowProxy;
    |     ^^^^^^^^^^^^^^^^^^^^^^^ no `WindowProxy` in `dom::types`
```

Comment (jdm) [inline components/script/dom/bindings/codegen/Bindings.conf]: Ah, ok!

**Pull request merged**